### PR TITLE
Update RLS and Rustfmt

### PR DESCRIPTION
**Pull request by nrc**
r? @Mark-Simulacrum 

Fixes tests

### CODE REVIEW

**Comment by kennytm** [comment]
@bors retry 

```
multipart upload failed: execution expired (Aws::S3::MultipartUploadError)
```

**Comment by pietroalbini** [comment]
Timeout downloading stuff.

@bors retry

**Comment by kennytm** [comment]
@bors treeclosed=10

The error in the last 18 hours seems to be systematic. Closing the tree until a successful merge is seen.

**Comment by kennytm** [comment]
(Note: Please don't retry if this failed again, I'm gonna roll this up)

File: src/Cargo.lock [hunk 1]
```diff
@@ -1699,10 +1700,10 @@ dependencies = [
  "rls-analysis 0.13.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "rls-blacklist 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
  "rls-data 0.16.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "rls-rustc 0.3.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rls-rustc 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "rls-span 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "rls-vfs 0.4.5 (registry+https://github.com/rust-lang/crates.io-index)",
- "rustfmt-nightly 0.8.2",
+ "rustfmt-nightly 0.8.2 (git+https://github.com/rust-lang-nursery/rustfmt?rev=f3906267)",
```
Comment (semarie) [inline src/Cargo.lock]: I am unsure if related, but currently I am unable to build beta (or nightly) from tarball using vendor=true due to network dependency (git) on `rustfmt`:

```
running: /usr/local/bin/cargo build --manifest-path /data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/Cargo.toml --frozen
    Updating git repository `https://github.com/rust-lang-nursery/rustfmt`
error: failed to load source for a dependency on `rustfmt-nightly`

Caused by:
  Unable to update https://github.com/rust-lang-nursery/rustfmt?rev=f3906267#f3906267

Caused by:
  failed to fetch into /data/semarie/build-rust/install_dir/crates/git/db/rustfmt-5390e0ead582d971

Caused by:
  attempting to update a git repository, but --frozen was specified
Traceback (most recent call last):
  File "/data/semarie/build-rust/build_dir/rustc-beta-src/x.py", line 20, in <module>
    bootstrap.main()
  File "/data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/bootstrap.py", line 827, in main
    bootstrap(help_triggered)
  File "/data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/bootstrap.py", line 806, in bootstrap
    build.build_bootstrap()
  File "/data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/bootstrap.py", line 633, in build_bootstrap
    run(args, env=env, verbose=self.verbose)
  File "/data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/bootstrap.py", line 148, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /usr/local/bin/cargo build --manifest-path /data/semarie/build-rust/build_dir/rustc-beta-src/src/bootstrap/Cargo.toml --frozen
```

**Pull request merged**